src/Illuminate/Foundation/Console/Kernel.php

```diff
@@ -294,6 +299,33 @@ protected function bootstrappers()
     }
 
     /**
+     * Set environment if 'env' option if provided.
+     *
+     * @param  InputInterface  $input
+     * @return void
+     */
+    protected function setEnvironment($input)
+    {
+        if (! app()->runningInConsole()) {
```

Comment:
I mean to say if i use `$this->app`, i can't access the following methods:

* runningInConsole
* environmentFile
* environmentPath
* loadEnvironmentFrom

If i use `app`, i can access the above listed methods.